src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1131,6 +1132,8 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 								"To proceed, either drop the distributed tables or use "
 								"undistribute_table() function to convert them to local tables")));
 		}
+
+		DeleteWorkerTransactions(workerNode);
```

Comment:
could be nice to have a comment on why we are calling this only for primary nodes